File: dailyQuestion/2020-09/9-30/python/solution.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def listOfDepth(tree: TreeNode) -> [ListNode]:
    result = []
    if not tree :
        return result
    queue = [tree]
    while len(queue) > 0 :
        size = len(queue)
        listnode = ListNode(0)
        tmp = listnode
        while size > 0 :
            node = queue[0]
            queue = queue[1:]
            tmp.next = ListNode(node.val)
            tmp = tmp.next
            if node.left :
                queue.append(node.left)
            if node.right :
                queue.append(node.right)
            size -= 1
        result.append(listnode.next)
    return result


if __name__ == "__main__" :
    node = TreeNode(3)
    node.left = TreeNode(5)
    node.right = TreeNode(1)
    node.left.left = TreeNode(6)
    node.left.right = TreeNode(2)
    node.left.right.left = TreeNode(7)
    node.left.right.right = TreeNode(4)

    node.right.left = TreeNode(0)
    node.right.right = TreeNode(8)

    result = listOfDepth(node)
    print(result)

File: dailyQuestion/2020-09/9-30/golang/solution.go
package main

import (
	"fmt"
)

func main() {
	root := &TreeNode{Val: 3, Left: &TreeNode{Val: 5}, Right: &TreeNode{Val: 1}}
	root.Left.Left = &TreeNode{Val: 6}
	root.Left.Right = &TreeNode{Val: 2, Left: &TreeNode{Val: 7}, Right: &TreeNode{Val: 4}}
	root.Right.Left = &TreeNode{Val: 0}
	root.Right.Right = &TreeNode{Val: 8}
	result := listOfDepth(root)
	fmt.Println(result)
}

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}
type ListNode struct {
	Val  int
	Next *ListNode
}

func listOfDepth(tree *TreeNode) []*ListNode {
	if tree == nil {
		return nil
	}
	result := []*ListNode{}
	queue := []*TreeNode{tree}
	for len(queue) > 0 {
		size := len(queue)
		listnode := &ListNode{Val: 0}
		tmp := listnode
		for i := 0; i < size; i++ {
			node := queue[0]
			queue = queue[1:]
			tmp.Next = &ListNode{Val: node.Val}
			tmp = tmp.Next
			if node.Left != nil {
				queue = append(queue, node.Left)
			}
			if node.Right != nil {
				queue = append(queue, node.Right)
			}
		}
		result = append(result, listnode.Next)
	}
	return result
}
